src/main/java/de/tum/in/www1/artemis/web/rest/competency/CourseCompetencyResource.java

```diff
@@ -53,4 +168,262 @@ public ResponseEntity<List<CourseCompetency>> getCourseCompetenciesWithProgress(
         final var competencies = courseCompetencyService.findCourseCompetenciesWithProgressForUserByCourseId(courseId, user.getId());
         return ResponseEntity.ok(competencies);
     }
+
+    /**
+     * GET courses/:courseId/course-competencies/:competencyId/student-progress gets the course competency progress for a user
+     *
+     * @param courseId     the id of the course to which the competency belongs
+     * @param competencyId the id of the course competency for which to get the progress
+     * @param refresh      whether to update the student progress or fetch it from the database (default)
+     * @return the ResponseEntity with status 200 (OK) and with the competency course performance in the body
+     */
+    @GetMapping("courses/{courseId}/course-competencies/{competencyId}/student-progress")
+    @EnforceAtLeastStudentInCourse
+    public ResponseEntity<CompetencyProgress> getCompetencyStudentProgress(@PathVariable long courseId, @PathVariable long competencyId,
+            @RequestParam(defaultValue = "false") boolean refresh) {
+        log.debug("REST request to get student progress for competency: {}", competencyId);
+        var user = userRepository.getUserWithGroupsAndAuthorities();
+        var course = courseRepository.findByIdElseThrow(courseId);
+        var competency = courseCompetencyRepository.findByIdElseThrow(competencyId);
+        checkCourseForCompetency(course, competency);
+
+        CompetencyProgress studentProgress;
+        if (refresh) {
+            studentProgress = competencyProgressService.updateCompetencyProgress(competencyId, user);
+        }
+        else {
+            studentProgress = competencyProgressRepository.findEagerByCompetencyIdAndUserId(competencyId, user.getId()).orElse(null);
+        }
+
+        return ResponseEntity.ok().body(studentProgress);
```

Comment:
Could we instead write
```suggestion
        return ResponseEntity.ok(studentProgress);
```